app/scripts/modules/core/src/delivery/status/ExecutionStatus.tsx

```diff
@@ -1,8 +1,135 @@
+import * as React from 'react';
+import * as ReactGA from 'react-ga';
+import { get, has } from 'lodash';
+import { BindAll } from 'lodash-decorators';
+
+import { IBuildTrigger, ICronTrigger, IDockerTrigger, IExecution } from 'core/domain';
+import { IScheduler } from 'core/scheduler/scheduler.factory';
+import { ReactInjector } from 'core/reactShims';
+import { relativeTime, timestamp } from 'core/utils';
+
+import { buildDisplayName } from '../executionBuild/buildDisplayName.filter';
+import { ExecutionBuildNumber } from '../executionBuild/ExecutionBuildNumber';
+
+import './executionStatus.less';
 
 export interface IExecutionStatusProps {
   execution: IExecution;
   toggleDetails: (stageIndex?: number) => void;
   showingDetails: boolean;
   standalone: boolean;
 }
+
+export interface IExecutionStatusState {
+  sortFilter: any;
+  parameters: { key: string, value: any }[];
+  timestamp: string;
+}
+
+@BindAll()
+export class ExecutionStatus extends React.Component<IExecutionStatusProps, IExecutionStatusState> {
+  private timestampScheduler: IScheduler;
+
+  constructor(props: IExecutionStatusProps) {
+    super(props);
+
+    // these are internal parameters that are not useful to end users
+    const strategyExclusions = [
+      'parentPipelineId',
+      'strategy',
+      'parentStageId',
+      'deploymentDetails',
+      'cloudProvider'
+    ];
+
+    let parameters: { key: string, value: any }[] = [];
+
+    const { execution } = this.props;
+    if (execution.trigger && execution.trigger.parameters) {
+      parameters = Object.keys(execution.trigger.parameters).sort()
+        .filter((paramKey) => execution.isStrategy ? !strategyExclusions.includes(paramKey) : true)
+        .map((paramKey: string) => {
+          return { key: paramKey, value: execution.trigger.parameters[paramKey] };
+        });
+    }
+
+    this.state = {
+      sortFilter: ReactInjector.executionFilterModel.asFilterModel.sortFilter,
+      parameters,
+      timestamp: relativeTime(this.props.execution.startTime),
+    }
+  }
+
+  private validateTimestamp(): void {
+    const newTimestamp = relativeTime(this.props.execution.startTime);
+    if (newTimestamp !== this.state.timestamp) {
+      this.setState({timestamp: newTimestamp});
+    }
+  }
+
+  public componentDidMount(): void {
+    this.timestampScheduler = ReactInjector.schedulerFactory.createScheduler();
+    this.timestampScheduler.subscribe(this.validateTimestamp);
+  }
+
+  public componentWillUnmount(): void {
+    this.timestampScheduler.unsubscribe();
+  }
+
+  private getExecutionTypeDisplay(): String {
+    const trigger = this.props.execution.trigger;
+    switch (trigger.type) {
+      case 'jenkins': return 'Triggered Build';
+      case 'manual': return 'Manual Start';
+      case 'pipeline': return 'Pipeline';
+      case 'docker': return 'Docker Registry';
+      case 'cron': return (trigger as ICronTrigger).cronExpression;
+      default: return trigger.type;
```

Comment:
Not something to deal with in this PR, but looking at, this should be come from the trigger configuration instead of wedged into this component.

Reply:
Very good call.